packages/widgets/src/menu.ts

```diff
@@ -319,6 +344,23 @@ export class Menu extends Widget {
     }
   }
 
+  /**
+   * Close the menu completely.
+   */
+  closeMenu(): void {
+    // Bail if the menu is not attached.
+    if (!this.isAttached) {
+      return;
+    }
+
+    // Cancel the pending timers.
+    this._cancelOpenTimer();
+    this._cancelCloseTimer();
+
+    // Close the root menu before executing the command.
+    this.rootMenu.close();
+  }
```

Comment:
I'm not sure I'm familiar enough with the Menu class to adequately review this method.